dm/master/scheduler/scheduler.go

```diff
@@ -1777,3 +1934,208 @@ func (s *Scheduler) SetWorkerClientForTest(name string, mockCli workerrpc.Client
 		s.workers[name].cli = mockCli
 	}
 }
+
+// nolint:dupl
+func (s *Scheduler) observeLoadTask(ctx context.Context, etcdCli *clientv3.Client, rev int64) error {
+	var wg sync.WaitGroup
+	for {
+		loadTaskCh := make(chan ha.LoadTask, 10)
+		loadTaskErrCh := make(chan error, 10)
+		wg.Add(1)
+		// use ctx1, cancel1 to make sure old watcher has been released
+		ctx1, cancel1 := context.WithCancel(ctx)
+		go func() {
+			defer func() {
+				close(loadTaskCh)
+				close(loadTaskErrCh)
+				wg.Done()
+			}()
+			ha.WatchLoadTask(ctx1, etcdCli, rev+1, loadTaskCh, loadTaskErrCh)
+		}()
+		err := s.handleLoadTask(ctx1, loadTaskCh, loadTaskErrCh)
+		cancel1()
+		wg.Wait()
+
+		if etcdutil.IsRetryableError(err) {
```

Comment:
`if etcdutil.IsRetryableError(err)` is repeated in many functions, we might wrap it in another PR 😂 passing a closure to same retry logic.